### ISSUE TITLE
[BUG] dynamicprompts 0.30.1 is broken

### ISSUE THREAD

**Issue by acorderob** [adieyal/sd-dynamic-prompts#665]
The new version of the library does not generate correct prompts. The result is all f****ed up with missing parts and partial statements.

Reverting to the commit prior to this change (e07ff1d72abfb1f9f4ae5b094e5e3ab173aeebde) fixes it.

**Comment by akx** [comment]
Can you be a bit more specific than "all f****ed up with missing parts and partial statements"? What is your original prompt? What are the results? Are there errors or warnings in the webui console?

**Comment by acorderob** [comment]
I just tested that wildcard by itself, and it fails.

The view.yaml file contains (among other wildcards):

```yaml
view:
  verticalangle_landscape:
    - "{3::|looking up view|looking down view|birds-eye view}"
```

And prompting with `__view/verticalangle_landscape__` results in

`|looking up view|looking down view|birds-eye view}`

So you should be able to reproduce.

**Comment by akx** [comment]
Ah. Well, that explains it – the `{3::` there conflicts with the new wildcard weight shorthand syntax introduced in https://github.com/adieyal/dynamicprompts/pull/97.

Turns out I was prescient with my comment there:

> The shorthand is not parsed from regular TXT files at this point, because I was afraid it might mess up someone's workflow. Somehow.

but not prescient enough, since this messed up a workflow even with a YAML wildcard file.

**Comment by acorderob** [comment]
Will you be able to fix it? or should i convert my wildcards to use the new weighted object syntax?

**Comment by akx** [comment]
I'll fix it, just a sec. 👍

**Comment by acorderob** [comment]
I feel like just a test that the string starts with "<number>::" would work. Maybe with a regex.

**Comment by akx** [comment]
Okay, this ought to be fixed with dynamicprompts 0.30.2 (uploading right now).

**Comment by akx** [comment]
Fixed with sddp 2.17.1 (current `main`).
